### DISCUSSION TITLE
Readiness Probe for uAgents

### DISCUSSION THREAD

**Discussion by jrriehl** [fetchai/uAgents#183] [uAgent Ideas]
# Readiness Probes

It could be useful in many cases to equip agents with a built-in health check protocol (https://github.com/fetchai/uAgents/issues/177).

**Proposal**: Agents should respond to requests of type

`HTTP 1.1 HEAD`

with headers:

```json
X-Uagents-Address: agent1qtrq7az22as8knqnwtyej7wmpgx0qsp0wzavunwxect0mkpnutjj6fpne3q
```

The response will indicate whether the agent is

1. **Ready**: alive and ready to handle envelopes. Additionally the response time hint header communicates to the client that the expected response time is 60 seconds (for example).

```json
HTTP/1.1 200 OK
X-Uagents-Status: ready
X-Uagents-Response-Time-Hint: 60

```

1. **Not Ready:** response from an endpoint shared by a collection of agents where the agent is not running or not responding at this time. 

Additionally the response time header is not set as there is no reasonable estimate for when a response can be expected from the agent

```json
HTTP/1.1 200 OK
X-Uagents-Status: not-ready
```

3. **Indeterminate**: response from an endpoint shared by a collection of agents where the health of the agent cannot be determined. This can be useful for a uagent mailbox where the target agent may or may not be known

Additionally the response time hint is set to a value that signals the sort of time frame. In this example it is 86400 seconds (24*60*60) or 24 hours:

```json
HTTP/1.1 200 OK
X-Uagents-Status: indeterminate
X-Uagents-Response-Time-Hint: 86400****
```

**Comment by ejfitzgerald** [comment]
Just for clarification `/submit` is an example and can be whatever endpoint is configured in the uAgent?

**Comment by Archento** [comment]
Closing this discussion as done since the feature is already implemented.
